### ISSUE TITLE
Dependabot does not create Grouped pull requests 

### ISSUE THREAD

**Issue by flash-me** [dependabot/dependabot-core#8552]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

npm, pnpm, yarn

### Package manager version

latest

### Language version

Node.js 20

### Manifest location and content before the Dependabot update

`frontend/package.json` Attached as [frontend-package.json](https://github.com/dependabot/dependabot-core/files/13590127/frontend-package.json)
`backend/package.json` Attaches as [backend-package.json](https://github.com/dependabot/dependabot-core/files/13590129/backend-package.json)

### dependabot.yml content

```
version: 2
updates:
  - package-ecosystem: npm
    directory: /frontend
    schedule:
      interval: daily
    versioning-strategy: auto
    # We do not want automatic major updates
    ignore:
      - dependency-name: "*"
        update-types: [ "version-update:semver-major" ]
    groups:
      # Group together updates for frontend
      frontend:
        patterns:
          - "*"

  - package-ecosystem: npm
    directory: /backend
    schedule:
      interval: daily
    versioning-strategy: auto
    # We do not want automatic major updates
    ignore:
      - dependency-name: "*"
        update-types: [ "version-update:semver-major" ]
    groups:
      # Group together updates for backend
      backend:
        patterns:
          - "*"

  - package-ecosystem: github-actions
    directory: /
    schedule:
      interval: weekly
```

### Expectation

We want a single **grouped pull request** for every dependency located in `/backend` and 
we also want another single **grouped pull request** for every dep located in `/frontend`

### Actual behaviour

Dependabot does not group any Pull Requests and creates a single PR for every dep update:

![SCR-20231206-qzyv](https://github.com/dependabot/dependabot-core/assets/5240450/23952077-64e2-4b06-a4eb-eb1dacd90a8f)

### Smallest manifest that reproduces the issue

```
version: 2
updates:
  - package-ecosystem: npm
    directory: /frontend
    schedule:
      interval: daily
    versioning-strategy: auto
    # We do not want automatic major updates
    ignore:
      - dependency-name: "*"
        update-types: [ "version-update:semver-major" ]
    groups:
      # Group together updates for frontend
      frontend:
        patterns:
          - "*"

  - package-ecosystem: npm
    directory: /backend
    schedule:
      interval: daily
    versioning-strategy: auto
    # We do not want automatic major updates
    ignore:
      - dependency-name: "*"
        update-types: [ "version-update:semver-major" ]
    groups:
      # Group together updates for backend
      backend:
        patterns:
          - "*"
```

**Comment by OwenPattison** [comment]
I also see the same behavior - in a slight variation to this.

As seen below, the group should update 9 dependencies but actually the pr shows only 1 of the 9 is actually updated.

![image](https://github.com/dependabot/dependabot-core/assets/5864244/70df37d2-9f82-409a-be4b-a13362bae024)

![image](https://github.com/dependabot/dependabot-core/assets/5864244/9fb645ba-1760-48f0-a8ab-b1492a2eab2e)

**Comment by gioce90** [comment]
I have the same problem, but I'm using Azure DevOps pipeline and NuGet.
All works, but instead of having 2 different PRs I see a PR for each dependency who needs an update.

**Comment by flash-me** [comment]
@Nishnha this is not only related to `dotnet:nuget`. We are having these issues with `npm`

**Comment by abdulapopoola** [comment]
We've shipped multiple fixes and this should be addressed; could you let us know if this is still happening for your repos?

**Comment by OwenPattison** [comment]
> We've shipped multiple fixes and this should be addressed; could you let us know if this is still happening for your repos?

Thanks for the update @abdulapopoola, were the fixes pushed at the time/date of this comment?

I ask because I still see the same behavior on a pull request opened on the 4th of March 2024.

I will monitor further grouped updates as they come and give feedback if I see a behavior change.

**Comment by abdulapopoola** [comment]
Thanks for confirming @OwenPattison ! The crew has been shipping multiple improvements every single day! Closing this as resolved.

**Comment by OwenPattison** [comment]
@abdulapopoola Unfortunately the next grouped update has come in with not-as-good news.

![image](https://github.com/dependabot/dependabot-core/assets/5864244/2b39c416-f2e5-415a-99bc-c887abc75837)

My group pattern looks like this:

```yaml      
      patterns:
        - "Microsoft*.EntityFrameworkCore*"
```
But I have noticed when confirming all references are updated that one of my projects with two dependencies that match this pattern has not been picked up and updated as part of this group as seen below.

![image](https://github.com/dependabot/dependabot-core/assets/5864244/b1430a67-aa8e-4007-b163-bf318b939d90)

To add a little more context this group update has targeted multiple projects however, it seems to have missed this one.

**Comment by abdulapopoola** [comment]
Thanks @OwenPattison ; could you please file a new issue with these details?

**Comment by flash-me** [comment]
Not sure why this is closed. I still have the same behaviour when I created the issue. 
<img width="958" alt="image" src="https://github.com/dependabot/dependabot-core/assets/5240450/de8a1695-c024-4f2f-aa46-ccd992ba62a3">

@abdulapopoola

**Comment by flash-me** [comment]
@abdulapopoola Also looking at the Update Job Details of Dependabot, they look very strange.
Even though I do have declared a group:

```yaml
updates:
    # .... 
    groups:
      # Group together updates for frontend
      frontend:
        patterns:
          - "*"
```

Both of the update jobs from yesterday have in their job configuration:

```
"dependency-groups": [],
```

### See below!

<details> 
  <summary> Job configuration one </summary>

```
{
  "job": {
    "allowed-updates": [
      {
        "dependency-type": "direct",
        "update-type": "all"
      }
    ],
    "commit-message-options": {
      "include-scope": null,
      "prefix": null,
      "prefix-development": null
    },
    "credentials-metadata": [
      {
        "host": "github.com",
        "type": "git_source"
      }
    ],
    "debug": null,
    "dependencies": [
      "@backstage/core-app-api"
    ],
    "dependency-group-to-refresh": null,
    "dependency-groups": [],
    "existing-group-pull-requests": [],
    "existing-pull-requests": [
      [
        {
          "dependency-name": "@testing-library/react",
          "dependency-version": "14.0.0"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.16.7"
        }
      ],
      [
        {
          "dependency-name": "@backstage/integration-react",
          "dependency-version": "1.1.13"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.16.17"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-common",
          "dependency-version": "0.7.6"
        }
      ],
      [
        {
          "dependency-name": "@backstage/dev-utils",
          "dependency-version": "1.0.19"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-common",
          "dependency-version": "0.19.3"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-node",
          "dependency-version": "0.7.12"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-tasks",
          "dependency-version": "0.5.6"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.17.14"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-app-backend",
          "dependency-version": "0.3.51"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-auth-backend",
          "dependency-version": "0.19.0"
        }
      ],
      [
        {
          "dependency-name": "@backstage/core-app-api",
          "dependency-version": "1.10.0"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-react",
          "dependency-version": "0.4.15"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-tasks",
          "dependency-version": "0.5.8"
        }
      ],
      [
        {
          "dependency-name": "@types/dockerode",
          "dependency-version": "3.3.20"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-tasks",
          "dependency-version": "0.5.11"
        }
      ],
      [
        {
          "dependency-name": "winston",
          "dependency-version": "3.11.0"
        }
      ],
      [
        {
          "dependency-name": "@backstage/theme",
          "dependency-version": "0.4.4"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-common",
          "dependency-version": "0.7.10"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-auth-node",
          "dependency-version": "0.4.2"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-node",
          "dependency-version": "0.7.19"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.19.4"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.19.6"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-tasks",
          "dependency-version": "0.5.14"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-react",
          "dependency-version": "0.4.19"
        }
      ],
      [
        {
          "dependency-name": "@backstage/core-app-api",
          "dependency-version": "1.11.3"
        }
      ],
      [
        {
          "dependency-name": "@backstage/dev-utils",
          "dependency-version": "1.0.26"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-auth-backend",
          "dependency-version": "0.20.3"
        }
      ],
      [
        {
          "dependency-name": "@types/express-serve-static-core",
          "dependency-version": "4.17.42"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.19.10"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-app-backend",
          "dependency-version": "0.3.60"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-common",
          "dependency-version": "0.21.4"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.19.26"
        }
      ]
    ],
    "experiments": {
      "proxy-cached": true,
      "record-ecosystem-versions": true,
      "record-update-job-unknown-error": true
    },
    "ignore-conditions": [
      {
        "dependency-name": "*",
        "source": ".github/dependabot.yaml",
        "update-types": [
          "version-update:semver-major"
        ],
        "updated-at": "2023-04-27T09:11:10.000Z",
        "version-requirement": null
      }
    ],
    "lockfile-only": false,
    "max-updater-run-time": 2700,
    "package-manager": "npm_and_yarn",
    "proxy-log-response-body-on-auth-failure": true,
    "reject-external-code": false,
    "repo-private": true,
    "requirements-update-strategy": null,
    "security-advisories": [],
    "security-updates-only": false,
    "source": {
      "api-endpoint": "https://api.github.com/",
      "branch": null,
      "directory": "/frontend",
      "hostname": "github.com",
      "provider": "github",
      "repo": "<REDACTED>"
    },
    "update-subdependencies": false,
    "updating-a-pull-request": true,
    "vendor-dependencies": false
  }
}
```

</details>


<details> 
  <summary> Job configuration two </summary>

```json
{
  "job": {
    "allowed-updates": [
      {
        "dependency-type": "direct",
        "update-type": "all"
      }
    ],
    "commit-message-options": {
      "include-scope": null,
      "prefix": null,
      "prefix-development": null
    },
    "credentials-metadata": [
      {
        "host": "github.com",
        "type": "git_source"
      }
    ],
    "debug": null,
    "dependencies": [
      "@backstage/dev-utils"
    ],
    "dependency-group-to-refresh": null,
    "dependency-groups": [],
    "existing-group-pull-requests": [],
    "existing-pull-requests": [
      [
        {
          "dependency-name": "@testing-library/react",
          "dependency-version": "14.0.0"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.16.7"
        }
      ],
      [
        {
          "dependency-name": "@backstage/integration-react",
          "dependency-version": "1.1.13"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.16.17"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-common",
          "dependency-version": "0.7.6"
        }
      ],
      [
        {
          "dependency-name": "@backstage/dev-utils",
          "dependency-version": "1.0.19"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-common",
          "dependency-version": "0.19.3"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-node",
          "dependency-version": "0.7.12"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-tasks",
          "dependency-version": "0.5.6"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.17.14"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-app-backend",
          "dependency-version": "0.3.51"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-auth-backend",
          "dependency-version": "0.19.0"
        }
      ],
      [
        {
          "dependency-name": "@backstage/core-app-api",
          "dependency-version": "1.10.0"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-react",
          "dependency-version": "0.4.15"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-tasks",
          "dependency-version": "0.5.8"
        }
      ],
      [
        {
          "dependency-name": "@types/dockerode",
          "dependency-version": "3.3.20"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-tasks",
          "dependency-version": "0.5.11"
        }
      ],
      [
        {
          "dependency-name": "winston",
          "dependency-version": "3.11.0"
        }
      ],
      [
        {
          "dependency-name": "@backstage/theme",
          "dependency-version": "0.4.4"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-common",
          "dependency-version": "0.7.10"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-auth-node",
          "dependency-version": "0.4.2"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-node",
          "dependency-version": "0.7.19"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.19.4"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.19.6"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-tasks",
          "dependency-version": "0.5.14"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-permission-react",
          "dependency-version": "0.4.19"
        }
      ],
      [
        {
          "dependency-name": "@backstage/core-app-api",
          "dependency-version": "1.11.3"
        }
      ],
      [
        {
          "dependency-name": "@backstage/dev-utils",
          "dependency-version": "1.0.26"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-auth-backend",
          "dependency-version": "0.20.3"
        }
      ],
      [
        {
          "dependency-name": "@types/express-serve-static-core",
          "dependency-version": "4.17.42"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.19.10"
        }
      ],
      [
        {
          "dependency-name": "@backstage/plugin-app-backend",
          "dependency-version": "0.3.60"
        }
      ],
      [
        {
          "dependency-name": "@backstage/backend-common",
          "dependency-version": "0.21.4"
        }
      ],
      [
        {
          "dependency-name": "@types/node",
          "dependency-version": "18.19.26"
        }
      ],
      [
        {
          "dependency-name": "@backstage/core-app-api",
          "dependency-version": "1.12.2"
        }
      ]
    ],
    "experiments": {
      "proxy-cached": true,
      "record-ecosystem-versions": true,
      "record-update-job-unknown-error": true
    },
    "ignore-conditions": [
      {
        "dependency-name": "*",
        "source": ".github/dependabot.yaml",
        "update-types": [
          "version-update:semver-major"
        ],
        "updated-at": "2023-04-27T09:11:10.000Z",
        "version-requirement": null
      }
    ],
    "lockfile-only": false,
    "max-updater-run-time": 2700,
    "package-manager": "npm_and_yarn",
    "proxy-log-response-body-on-auth-failure": true,
    "reject-external-code": false,
    "repo-private": true,
    "requirements-update-strategy": null,
    "security-advisories": [],
    "security-updates-only": false,
    "source": {
      "api-endpoint": "https://api.github.com/",
      "branch": null,
      "directory": "/frontend",
      "hostname": "github.com",
      "provider": "github",
      "repo": "<REDACTED>"
    },
    "update-subdependencies": false,
    "updating-a-pull-request": true,
    "vendor-dependencies": false
  }
}
```

</details>


Do we have a wrong configuration?

**Comment by abdulapopoola** [comment]
Thanks @flash-me , will re-open this issue.

**Comment by jakecoffman** [comment]
If you notice that there are no dependency-groups in the job definition printed in the logs (`"dependency-groups": [],`) but you have groups defined in dependabot.yml, that either means:
1. The dependabot.yml is invalid/has some kind of error.
2. The Dependabot service missed the webhook from when dependabot.yml was updated and it needs to reprocess the config file.

If you think it's 2, then you can either make a whitespace change to dependabot.yml to get it to send another webhook, or link me to the repo (or give me a job ID from a Dependabot run in the repo so I can look it up) and I can have the service reprocess the config file.

**Comment by flash-me** [comment]
@jakecoffman @abdulapopoola

Managed to get it working. What a coincidence.

I closed every PR created by dependabot and then added the `applies-to` property in the `dependabot yaml` file.

But reading your comment, any kind of change would have been sufficient, I guess.

Nonetheless, for other people facing the same challenges, I suggest to provide / describe these "debugging" steps in the docs or similar places.

Thanks



cheers
flash ⚡
